### PR TITLE
templates(netlify): improve DX for Netlify serverless template

### PR DESCRIPTION
**Pull request by nickytonline**
This PR improves the DX when using the Netlify template to build out your Remix application. cc: @ascorbic. I've based this PR off of `main` because I believe @mcansh mentioned that if it's just template changes, there's no need to branch off of the `dev` branch.

Closes: #3753

- [ ] Docs
- [ ] Tests

Testing Strategy:

1. Create a new Remix application and select Netlify for the deployment target.
2. You can install whatever you'd like in the Remix CLI steps. It also doesn't matter if you select TypeScript or JavaScript
3. Once your application has been created, from a shell, change to the root directory of your new application.
4. Run `npm run dev`
5. The Remix dev server starts and renders the main page when navigating to http://localhost:3000
6. Stop the Remix dev server.
7. Run `netlify dev`. The server running on Netlify functions locally starts and renders the main page when navigating to http://localhost:3000.
8. Run `netlify deploy --build`. If you're application isn't configured to deploy yet, ensure you've followed the steps in the README first to get your app configured to deploy to Netlify.
9. The application builds and deploys. Navigate to the deploy preview. The site loads.

### CODE REVIEW

**Comment by nickytonline** [comment]
Before this gets merged, I'd like to have at least one other of my team members approve it, so for the moment, I will put it back in draft.

**Comment by machour** [comment]
@nickytonline all PR only gets merged when one of the team members approve it, so if you're done with it, you can safely remove the draft status 👍🏼

File: templates/netlify/README.md
Comment (ascorbic) [inline templates/netlify/README.md:49]: The wording here is a bit strange: "the serverless version of the server"!

File: templates/netlify/remix.config.js
Comment (ascorbic) [reply templates/netlify/remix.config.js:5]: Would a check for `process.env.NETLIFY` work instead?

Comment (nickytonline) [reply templates/netlify/remix.config.js:5]: TIL. That definitely sounds more robust. I'll go ahead and make the change. Thanks Matt!

File: templates/netlify/README.md
Comment (KyleBlankRollins) [inline templates/netlify/README.md:33]: [sand] Minor addition to make the connection between `npm run dev` and `remix dev` clearer.
```suggestion
The Remix dev server starts your app in development mode, rebuilding assets on file changes. To start the Remix dev server:
```

Comment (KyleBlankRollins) [inline templates/netlify/README.md:33]: [pebble] It may make sense to sequence the content in this section. Do users need to have the Remix App Server running before starting `netlify dev`?

Even if they don't, as currently written, we say the Remix server starts the app in dev mode. Then we way the Netlify CLI does the same thing. And we don't really define the relationship, so it could be interpreted as either or both running the app in a non-specific way.

```suggestion
To run your Remix app locally and properly test Netlify features, you need to:
1. Start the Remix App Server in development mode
2. Start Netlify Dev

### Start the Remix App Server in development mode

The Remix dev server starts your app in development mode, rebuilding assets on file changes. To start the Remix dev server:
```

File: templates/netlify/README.md
Comment (KyleBlankRollins) [inline templates/netlify/README.md:41]: [pebble] Continuation of sequencing suggestion from above.

```suggestion
### Start Netlify Dev

The Netlify CLI starts your app in development mode for testing the serverless server and any custom Netlify functions you've developed.
```

Comment (KyleBlankRollins) [inline templates/netlify/README.md:41]: [pebble] Suggested alternative wording to avoid "serverless server".

```suggestion
The Netlify CLI builds a production version of your Remix App Server and splits it into Netlify Functions that run locally. This includes any custom Netlify functions you've developed. The Netlify CLI runs all of this in its development mode.
```

File: templates/netlify/README.md
Comment (KyleBlankRollins) [inline templates/netlify/README.md:49]: [pebble] Suggested alternative wording to avoid "serverless server".
```suggestion
Note: When running the Netlify CLI, file changes will rebuild assets, but you will not see the changes to the page you are on unless you do a browser refresh of the page. Due to how the Netlify CLI builds the Remix App Server, it does not support hot module reloading.
```

**Review by KyleBlankRollins — COMMENTED** [review]
Thanks for tagging me, @nickytonline! Although I personally find "serverless server" hilarious, I'm not sure we want to use it in the docs.

Let me know if you have questions!

File: templates/netlify/README.md
Comment (nickytonline) [reply templates/netlify/README.md]: No in this case, when you run `netlify dev`, it will build the code to run the Remix app server in Netlify functions.

File: templates/netlify/remix.config.js
Comment (nickytonline) [inline templates/netlify/remix.config.js:4]: This check is to ensure we do not run the Netlify functions server when running `npm run dev`, i.e. `remix dev` as you can't run a custom server via `remix dev`. Without this check, if you run `remix dev` it will error out with the following message:

```
Error: remix dev is not supported for custom servers.
    at Object.dev (/Users/some_user/dev/issues/remix-tailwind/node_modules/@remix-run/dev/cli/commands.js:260:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.run (/Users/some_user/dev/issues/remix-tailwind/node_modules/@remix-run/dev/cli/run.js:499:7)
ERROR: "dev:remix" exited with 1.
```

**Review by MichaelDeBoey — APPROVED** [review]

**Review by kentcdodds — APPROVED** [review]
Thank you!

**Comment by devraj** [comment]
@nickytonline I was wondering if this was released? I still see something similar in `1.15` and the app ends up with hydration exceptions. The original issue is now locked hence posting here


Screenshot 2023-04-29 at 3 48 15 pm

**Comment by nickytonline** [comment]
@devraj, the home for the Netlify templates can now be found in this Remix stack, https://github.com/netlify/remix-template. The CLI still points to the one that only supports the Netlify Functions template.

Kent or others, I know long term, folks are encouraged to build their templates outside of the repo, but given that Netlify is still an option in the CLI like Vercel, Deno etc., could we point to the https://github.com/netlify/remix-template Stack instead when someone selects Netlify?

Happy to create a PR for this.

**Comment by machour** [comment]
@nickytonline Your best course of action here is to create a new proposal in the discussions tab 🙏🏼

**Comment by devraj** [comment]
> @devraj, the home for the Netlify templates can now be found in this Remix stack, https://github.com/netlify/remix-template. The CLI still points to the one that only supports the Netlify Functions template.
> 
> Kent or others, I know long term, folks are encouraged to build their templates outside of the repo, but given that Netlify is still an option in the CLI like Vercel, Deno etc., could we point to the https://github.com/netlify/remix-template Stack instead when someone selects Netlify?
> 
> Happy to create a PR for this.

I agree. If that repo is what is kept up to date, then it would save everyone deploying to Netlify a lot of time. Appreciate you taking the time to post a response..

**Comment by nickytonline** [comment]
@devraj, just FYI, the CLI is getting updated with the official template. See https://github.com/remix-run/remix/pull/6281

**Pull request merged**
